source/CMakeLists.txt

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required(VERSION 3.13)
+
+set(CMAKE_C_STANDARD 99)
+
+# Azure IoT FreeRTOS Middleware Library
+add_library(az_iot_middleware_freertos
+  ${CMAKE_CURRENT_LIST_DIR}/azure_iot_hub_client.c
+  ${CMAKE_CURRENT_LIST_DIR}/azure_iot_provisioning_client.c
+  ${CMAKE_CURRENT_LIST_DIR}/azure_iot.c
+)
+
+# The user needs to supply this
+target_link_libraries(az_iot_middleware_freertos
+  PUBLIC
+    middleware_config)
+
+target_link_libraries(az_iot_middleware_freertos
+  PUBLIC
+    az::iot::common
+    az::iot::hub
+    az::iot::provisioning
+)
+
+# Include port files
+# TODO: Have the user select the port path to use
+if(WIN32)
+  target_include_directories(az_iot_middleware_freertos
+    PRIVATE
+      ${az_iot_middleware_freertos_SOURCE_DIR}/libraries/FreeRTOS/FreeRTOS/Source/portable/MSVC-MingW
+  )
+elseif(UNIX)
+  target_include_directories(az_iot_middleware_freertos
+    PRIVATE
+      ${az_iot_middleware_freertos_SOURCE_DIR}/libraries/FreeRTOS/FreeRTOS/Source/portable/ThirdParty/GCC/Posix
+  )
+endif()
+
+# Include FreeRTOS headers
+target_include_directories(az_iot_middleware_freertos
+  PUBLIC
+    ${az_iot_middleware_freertos_SOURCE_DIR}/libraries/FreeRTOS/FreeRTOS/Source/include
+)
+
+target_include_directories(az_iot_middleware_freertos
+    PUBLIC
+      ${CMAKE_CURRENT_LIST_DIR}/include
+      ${CMAKE_CURRENT_LIST_DIR}/interface
+)
+
+# Check if custom mqtt port path is set, otherwise
+# use default coreMQTT port
+if(AZURE_IOT_MQTT_PORT)
```

Comment:
>AZURE_IOT_MQTT_PORT [](http://example.com/codeflow?start=3&length=19)

Make me think that having three target this lib compilation depends upon.

1, FreeRTOS
2. FreeRTOS port.
3. MQTT port.

and these target can be define by application. What you think?

Reply:
I think I fundamentally agree with you. Something like three paths for those three parts? Environment variables?